src/cfnlint/rules/resources/updatepolicy/Configuration.py

```diff
@@ -0,0 +1,221 @@
+                    'ValidValues': [
+                        'Launch', 'Terminate', 'HealthCheck',
+                        'ReplaceUnhealthy', 'AZRebalance', 'AlarmNotification',
+                        'ScheduledActions', 'AddToLoadBalancer'
+                    ]
+                },
+                'WaitOnResourceSignals': {
+                    'PrimitiveType': bool
+                }
+            },
+            'AutoScalingScheduledAction': {
+                'IgnoreUnmodifiedGroupSizeProperties': {
+                    'PrimitiveType': bool
+                }
+            },
+            'CodeDeployLambdaAliasUpdate': {
+                'AfterAllowTrafficHook': {
+                    'PrimitiveType': six.string_types
+                },
+                'ApplicationName': {
+                    'PrimitiveType': six.string_types
+                },
+                'BeforeAllowTrafficHook': {
+                    'PrimitiveType': six.string_types
+                },
+                'DeploymentGroupName': {
+                    'PrimitiveType': six.string_types
+                },
+            }
+        },
+        'main': {
+            'AutoScalingReplacingUpdate': {
+                'Type': 'AutoScalingReplacingUpdate',
+                'ResourceTypes': [
+                    'AWS::AutoScaling::AutoScalingGroup'
+                ]
+            },
+            'AutoScalingRollingUpdate': {
+                'Type': 'AutoScalingRollingUpdate',
+                'ResourceTypes': [
+                    'AWS::AutoScaling::AutoScalingGroup'
+                ]
+            },
+            'AutoScalingScheduledAction': {
+                'Type': 'AutoScalingScheduledAction',
+                'ResourceTypes': [
+                    'AWS::AutoScaling::AutoScalingGroup'
+                ]
+            },
+            'CodeDeployLambdaAliasUpdate': {
+                'Type': 'CodeDeployLambdaAliasUpdate',
+                'ResourceTypes': [
+                    'AWS::Lambda::Alias'
+                ]
+            },
+            'UseOnlineResharding': {
+                'PrimitiveType': bool,
+                'ResourceTypes': [
+                    'AWS::ElastiCache::ReplicationGroup'
+                ]
+            }
+        }
+    }
+
+    def check_primitive_type(self, value, prim_type, path, **kwargs):
+        """
+            Check primitive types
+        """
+        matches = []
+
+        if isinstance(value, dict):
+            # this could be a ref, getatt, findinmap
+            return matches
+
+        default_message = 'Value for {0} must be of type {1}'
+        if not isinstance(value, prim_type):
+            if prim_type == six.string_types:
+                s_type = 'String'
+            elif prim_type == int:
+                s_type = 'Integer'
+            elif prim_type == list:
+                s_type = 'List'
+            elif prim_type == bool:
+                s_type = 'Boolean'
```

Comment:
you could store the names in the dict too to avoid all these ifs - data not code :) Also I'd suggest `type_name` as the var name